lib/index.js

```diff
@@ -442,14 +453,25 @@ function compoundQueryParams (searchParams, queryCollection = []) {
 }
 
 /* Parse domains from operations or options */
-function parseServers (domains, serversOpts) {
+function parseServers (variables, domains, serversOpts) {
   let servers
   if (serversOpts != null) {
-    // This map is just to filter not supported fields while no validations are implemented
-    servers = serversOpts.map(({ url, description }) => ({ url, description }))
+    // This map is just to filter unsupported fields while no validations are implemented
+    servers = serversOpts.map(({ url, description, variables }) => ({ url, description, variables }))
   } else {
-    servers = Array.from(domains).map(domain => ({ url: domain }))
+    servers = Array.from(domains).map(([domain, variableNames]) => {
```

Comment:
As an aside, I noticed that because you're doing `new URL()` in `scrapeURL()` if an item URL has a port variable, like "https://{{name}}.example.com:{{port}}/:basePath/", the whole library crashes as that's not a valid URL. Might want to rethink how we're extracting that data out at some point but I think it's a non-issue at the moment.